packages/node-http-handler/src/node-http2-handler.spec.ts

```diff
@@ -28,288 +29,420 @@ describe("NodeHttp2Handler", () => {
+        // @ts-ignore: access private property
+        const session: ClientHttp2Session = nodeH2Handler.sessionCache.get(authority)[0];
+        expect(session.destroyed).toBe(false);
+        // @ts-ignore: access private property
+        expect(nodeH2Handler.sessionCache.get(authority).length).toStrictEqual(1);
+        setTimeout(() => {
+          expect(session.destroyed).toBe(true);
+          // @ts-ignore: access private property
+          expect(nodeH2Handler.sessionCache.get(authority).length).toStrictEqual(0);
+          done();
+        }, sessionTimeout + 100);
+      });
+
+      it("disableConcurrentStreams: true", async (done) => {
+        let session;
+        const authority = `${protocol}//${hostname}:${port}`;
+
+        nodeH2Handler = new NodeHttp2Handler({ sessionTimeout, disableConcurrentStreams: true });
+
+        mockH2Server.removeAllListeners("request");
+        mockH2Server.on("request", (request: any, response: any) => {
+          // @ts-ignore: access private property
+          session = nodeH2Handler.sessionCache.get(authority)[0];
+          createResponseFunction(mockResponse)(request, response);
+        });
+        await nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {});
+
+        expect(session.destroyed).toBe(false);
+        setTimeout(() => {
+          expect(session.destroyed).toBe(true);
+          done();
+        }, sessionTimeout + 100);
+      });
     });
   });
 
-  describe("abortSignal", () => {
-    it("will not create session if request already aborted", async () => {
-      // @ts-ignore: access private property
-      expect(nodeH2Handler.connectionPool.size).toBe(0);
-      await expect(
-        nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {
-          abortSignal: {
-            aborted: true,
-            onabort: null,
-          },
-        })
-      ).rejects.toHaveProperty("name", "AbortError");
-      // @ts-ignore: access private property
-      expect(nodeH2Handler.connectionPool.size).toBe(0);
+  describe("disableConcurrentStreams", () => {
+    beforeEach(() => {
+      nodeH2Handler = new NodeHttp2Handler({
+        disableConcurrentStreams: true,
+      });
     });
 
-    it("will not create request on session if request already aborted", async () => {
-      await nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {});
-
-      // @ts-ignore: access private property
-      const session: ClientHttp2Session = nodeH2Handler.connectionPool.get(`${protocol}//${hostname}:${port}`);
-      const requestSpy = jest.spyOn(session, "request");
-
-      await expect(
-        nodeH2Handler.handle(new HttpRequest(getMockReqOptions()), {
-          abortSignal: {
-            aborted: true,
-            onabort: null,
-          },
-        })
-      ).rejects.toHaveProperty("name", "AbortError");
-      expect(requestSpy.mock.calls.length).toBe(0);
+    describe("number calls to http2.connect", () => {
```

Comment:
duplicated to line 57?

Reply:
No. This section is for `disableConcurrentStreams: true`
The section on line 57 is for `disableConcurrentStreams: false`